### ISSUE TITLE
st2 services do not start on boot up

### ISSUE THREAD

**Issue by pilyk** [StackStorm/st2#2892]
I am running on Redhat 7 and ST2 1.6.0.

I have noticed that st2 services are not all up after a boot up.  However, if I perform a st2ctl restart, all the services are up.  My suspicion is that the st2 services are relying on mongo db and rabbitmq but they have not been completed started so the st2 services does not start.  I would need guidance to confirm this as well as the remedy. 

When executing st2ctl status after a boot up,  I see the following status:

st2actionrunner is not running.
st2api is not running.
st2stream is not running.
st2auth is not running.
st2garbagecollector PID: 2660
st2notifier is not running.
st2resultstracker PID: 1315
st2rulesengine PID: 1319
st2sensorcontainer PID: 1310
st2chatops is not running.
mistral-server PID: 1320
mistral-api PID: 1306
mistral-api PID: 2666
mistral-api PID: 2671

Thanks.

**Comment by pilyk** [comment]
I should have also mentioned that it is a all in one box installation of Stackstorm.  Thanks.

**Comment by luizsoares** [comment]
I had a similar issue where the culprit was mongo taking longer to start than the start timeout allowed. That happened because apparently on first boot mongo pre-initializes a bunch of things on disk and my root volume was not large enough to be granted a decent amount of IOPS by AWS.

If you're on AWS as well, try creating an instance with a really large disk.

**Comment by pilyk** [comment]
@luizsoares Thanks for the suggestion.  However, I am using internal resources and not AWS or third party.  The problem still might be due to lack of resource.  I was hoping to leverage systemd unit files to handle the dependency issue but so far no solution yet.

**Comment by LindsayHill** [comment]
I seem to recall @DoriftoShoes thought it might be related to needing to figure out dependencies between services starting with systemd vs sysv-style

**Comment by pilyk** [comment]
To give you the errors I am encountering, the below is a snippet of the st2api.log:

2016-09-09 11:38:02,197 38729520 WARNING coordination [-] Coordination backend is not configured. Code paths which use coordination se
rvice will use best effort approach and race conditions are possible.
2016-09-09 11:38:02,227 38729520 INFO app [-] st2api.app app created.
2016-09-09 11:38:02,232 63490320 INFO webhooks [-] Listening to endpoint: /webhooks/sample
2016-09-09 11:38:02,234 63490160 INFO mixins [-] Connected to amqp://guest:**@127.0.0.1:5672//
2016-09-09 11:44:13,912 63490160 WARNING mixins [-] Connection to broker lost, trying to re-establish connection...
Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/mixins.py", line 177, in run
    for _ in self.consume(limit=None):  # pragma: no cover
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/mixins.py", line 199, in consume
    conn.drain_events(timeout=safety_interval)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/connection.py", line 275, in drain_events
    return self.transport.drain_events(self.connection, **kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/transport/pyamqp.py", line 95, in drain_events
    return connection.drain_events(**kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/connection.py", line 324, in drain_events
    return amqp_method(channel, args)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/connection.py", line 527, in _close
    self._x_close_ok()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/connection.py", line 555, in _x_close_ok
    self._send_method((10, 51))
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/abstract_channel.py", line 56, in _send_method
    self.channel_id, method_sig, args, content,
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/method_framing.py", line 221, in write_method
    write_frame(1, channel, payload)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/transport.py", line 182, in write_frame
    frame_type, channel, size, payload, 0xce,
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/eventlet/greenio/base.py", line 385, in sendall
    tail = self.send(data, flags)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/eventlet/greenio/base.py", line 379, in send
    return self._send_loop(self.fd.send, data, flags)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/eventlet/greenio/base.py", line 366, in _send_loop
    return send_method(data, *args)
error: [Errno 104] Connection reset by peer
2016-09-09 11:44:13,962 63490160 WARNING mixins [-] Broker connection error, trying again in 2.0 seconds: error(error(111, 'ECONNREFUSED'),).
Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/utils/__init__.py", line 246, in retry_over_time
    return fun(*args, **kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/connection.py", line 237, in connect
    return self.connection
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/connection.py", line 742, in connection
    self._connection = self._establish_connection()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/connection.py", line 697, in _establish_connection
    conn = self.transport.establish_connection()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/kombu/transport/pyamqp.py", line 116, in establish_connection
    conn = self.Connection(**opts)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/connection.py", line 165, in __init__
    self.transport = self.Transport(host, connect_timeout, ssl)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/connection.py", line 186, in Transport
    return create_transport(host, connect_timeout, ssl)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/transport.py", line 299, in create_transport
    return TCPTransport(host, connect_timeout)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/amqp/transport.py", line 95, in __init__
    raise socket.error(last_err)
error: [Errno 111] ECONNREFUSED

Could it be a rabbitmq service issue?

**Comment by pilyk** [comment]
@LindsayHill You have a good point.  I know that mongod is a SysV but reading about systemd, it has a generator to create .service wrapper for it.  So not sure.

**Comment by LindsayHill** [comment]
@pilyk I've been avoiding getting too close to systemd, but it's probably time...

**Comment by pilyk** [comment]
I have also looked at the result of a systemd-analyze and rabbitmq takes 6.5 seconds of time which relatively is quite a bit.

**Comment by pilyk** [comment]
Yes, I have been looking into systemd more than I hoped for :)  Still not having much positive results yet.  I have tried the [Timer] and OnStartupSec=60 and that didn't help.  So I am a bit of a loss right now.

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/community/p1474048936001646

**Comment by bigmstone** [comment]
I've verified this bug on centos7.

```
[vagrant@localhost ~]$ sudo st2ctl status
##### st2 components status #####
st2actionrunner is not running.
st2api is not running.
st2stream is not running.
st2auth is not running.
st2garbagecollector PID: 918
st2notifier is not running.
st2resultstracker PID: 924
st2rulesengine PID: 917
st2sensorcontainer PID: 913
st2chatops is not running.
mistral-server PID: 916
mistral-api PID: 919
mistral-api PID: 1445
mistral-api PID: 1450
```

**Comment by samirsss** [comment]
This was seen on an ubuntu as well in the CID lab last night. Not sure if its related but just wanted to update it here since its similar in nature.

sudo st2ctl restart fixed the issue.

**Comment by m4dcoder** [comment]
I was able to reproduce this issue on a centos7 vagrant box. If you run `sudo systemctl enable` for all the services (i.e. `sudo systemctl enable st2auth`), it should fix the problem. Please try this and confirm.

**Comment by m4dcoder** [comment]
This line for packaging seems to be broken.  https://github.com/StackStorm/st2-packages/blob/master/packages/st2/rpm/st2.spec#L34

**Comment by pilyk** [comment]
Great!  That did it.  I had to systemctl enable the st2 services.  Specifically, I had to enable the following:

st2api.service
st2auth.service
st2stream.service
st2actionrunner.service
st2notifier.service

Thanks for all your help.

**Comment by m4dcoder** [comment]
This is fixed. It will be included in the v2.0.1 patch when it's released. https://github.com/StackStorm/st2-packages/pull/344

**Comment by pkankar1** [comment]
Hello guys,
I installed st2 HA. All my pods are in running state. But, when I execute st2ctl reload from inside of st2client pods, the status of the services are shown as not working.
```
root@happy-labradoodle-st2client-5bc549bc48-fmjvm:/etc/default# st2ctl reload
Registering content...[flags = --config-file /etc/st2/st2.conf --config-file=/etc/st2/st2.docker.conf --config-file=/etc/st2/st2.user.conf --register-runners --register-actions --register-aliases --register-sensors --register-triggers --register-configs --register-rules]
2020-07-18 17:41:40,351 INFO [-] Connecting to database "st2" @ "happy-labradoodle-mongodb-ha-0.happy-labradoodle-mongodb-ha:27017,happy-labradoodle-mongodb-ha-1.happy-labradoodle-mongodb-ha:27017,happy-labradoodle-mongodb-ha-2.happy-labradoodle-mongodb-ha:27017 (replica set)" as user "admin".
2020-07-18 17:41:40,366 INFO [-] Successfully connected to database "st2" @ "happy-labradoodle-mongodb-ha-0.happy-labradoodle-mongodb-ha:27017,happy-labradoodle-mongodb-ha-1.happy-labradoodle-mongodb-ha:27017,happy-labradoodle-mongodb-ha-2.happy-labradoodle-mongodb-ha:27017 (replica set)" as user "admin".
2020-07-18 17:41:40,853 INFO [-] =========================================================
2020-07-18 17:41:40,853 INFO [-] ############## Registering triggers #####################
2020-07-18 17:41:40,853 INFO [-] =========================================================
2020-07-18 17:41:40,952 INFO [-] Registered 0 triggers.
2020-07-18 17:41:40,952 INFO [-] =========================================================
2020-07-18 17:41:40,952 INFO [-] ############## Registering sensors ######################
2020-07-18 17:41:40,952 INFO [-] =========================================================
2020-07-18 17:41:40,979 INFO [-] Registered 1 sensors.
2020-07-18 17:41:40,979 INFO [-] =========================================================
2020-07-18 17:41:40,979 INFO [-] ############## Registering runners ######################
2020-07-18 17:41:40,979 INFO [-] =========================================================
2020-07-18 17:41:41,920 INFO [-] Registered 14 runners.
2020-07-18 17:41:41,921 INFO [-] =========================================================
2020-07-18 17:41:41,921 INFO [-] ############## Registering actions ######################
2020-07-18 17:41:41,921 INFO [-] =========================================================
2020-07-18 17:41:43,589 INFO [-] Registered 73 actions.
2020-07-18 17:41:43,589 INFO [-] =========================================================
2020-07-18 17:41:43,589 INFO [-] ############## Registering rules ########################
2020-07-18 17:41:43,589 INFO [-] =========================================================
2020-07-18 17:41:43,721 INFO [-] Registered 3 rules.
2020-07-18 17:41:43,721 INFO [-] =========================================================
2020-07-18 17:41:43,721 INFO [-] ############## Registering aliases ######################
2020-07-18 17:41:43,721 INFO [-] =========================================================
2020-07-18 17:41:43,796 INFO [-] Registered 4 aliases.
2020-07-18 17:41:43,796 INFO [-] =========================================================
2020-07-18 17:41:43,797 INFO [-] ############## Registering configs ######################
2020-07-18 17:41:43,797 INFO [-] =========================================================
2020-07-18 17:41:43,815 INFO [-] Registered 2 configs.
##### st2 components status #####
st2actionrunner is not running.
st2api is not running.
st2stream is not running.
st2auth is not running.
st2garbagecollector is not running.
st2notifier is not running.
st2resultstracker is not running.
st2rulesengine is not running.
st2sensorcontainer is not running.
st2chatops is not running.
st2timersengine is not running.
st2workflowengine is not running.
st2scheduler is not running.
```
I tried doing systemctl enable <services>, but it did not work.
Can someone help me. TIA
